src/main/java/graphql/schema/DataFetcher.java

```diff
@@ -3,5 +3,5 @@
 
 public interface DataFetcher {
```

Comment:
so then `#dataFetcher(DataFetcher fetcher)` becomes `#<T>dataFetcher(DataFetcher<T> fetcher)` and then inside `ExecutionStrategy#resolveField()` you get something like `T result = fetcher.get(...)` and then all the `completeValue()` signatures change. Do you really get much added type safety from this?